apps/prairielearn/src/lib/question-render.ts

```diff
@@ -495,7 +550,7 @@ export async function getAndRenderVariant(variant_id, variant_seed, locals) {
 
   if (locals.instance_question) {
     await manualGrading.populateRubricData(locals);
-    await async.each(locals.submissions, manualGrading.populateManualGradingData);
+    await async.eachSeries(locals.submissions, manualGrading.populateManualGradingData);
```

Comment:
`populateManualGradingData` talks to the database; we really don't want to allow unbounded database concurrency for a single request.

Reply:
I messaged on Slack, but I'll comment here for posterity: I can't see where `populateManualGradingData` ever does a query for rubric data. It does select rubric *grading* data, but that's a distinct thing that's querying `rubric_grading_items` and `rubric_gradings`. What am I missing here?

Reply:
There is a separate `populateRubricData` function in manualGrading.ts.
https://github.com/PrairieLearn/PrairieLearn/blob/9437c73dc196f94ecda312596880e540f427912b/apps/prairielearn/src/lib/manualGrading.ts#L125

Reply:
@jonatanschroeder this PR changed that to be a simple wrapper around the new `selectRubricData` function. Regardless, I don't see how that's relevant to Matt's comment?

Reply:
Oh yes, apologies, @jonatanschroeder was correct that I was confusing the `populateRubricData()` function that was edited elsewhere in this PR with the `populateManualGradingData()` function that's being called here.

So we are back to the fact that this is just a regular N+1 query problem, and not an even worse problem as I had thought 😅.